### PR TITLE
Add Dockerfile and deployment script for Functions

### PR DESCRIPTION
**Pull request by tiziano88**
For now it is a combined image with both the loader and the actual
application, but in the future those will be unbundled.

Ref #1934

# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: scripts/deploy_oak_functions_loader [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+docker build \
+  --tag="${FUNCTIONS_DOCKER_IMAGE_NAME}:latest" \
+  ./oak_functions
+
+docker push "${FUNCTIONS_DOCKER_IMAGE_NAME}:latest"
+
+readonly FUNCTIONS_INSTANCE_NAME=oak-functions-example
+
+gcloud auth activate-service-account \
+  --project="${GCP_PROJECT_ID}" \
```
Comment (rbehjati) [inline scripts/deploy_oak_functions_loader:24]: Maybe add a brief comment that these are environment variables that should be set on the system.

File: oak_functions/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/*
+!/loader/target/x86_64-unknown-linux-musl/release/oak_functions_loader
+!/loader/testdata/non-oak-minimal.wasm
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply oak_functions/.dockerignore:3]: Exactly, which is why it is un-ignored here, right?

File: oak_functions/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# TODO(#1934): Separate the base loader image from that of individual examples.
+COPY ./loader/testdata/non-oak-minimal.wasm ${wasm_module_path}
+
+# We cannot use ARG values as part of ENTRPOINT, so we have to repeat them here.
+# https://docs.docker.com/engine/reference/builder/#environment-replacement
+#
+# Normally we would use the $PORT env variable here, but since we do not have a shell available in
+# the Docker image, we hardcode its default value (8080) here.
+# https://cloud.google.com/run/docs/reference/container-contract#port
+ENTRYPOINT ["/oak_functions_loader", "--http-listen-port=8080", "--wasm-path=/module.wasm"]
```
[the remark this answers is not in the document]
Comment (tiziano88) [reply oak_functions/Dockerfile:17]: Sadly it does not work, as per the comment above :/

File: scripts/deploy_oak_functions_loader (hunk 1 above)
Comment (tiziano88) [reply scripts/deploy_oak_functions_loader:24]: I can add it, though these are already defined and checked by the included gcp script

File: oak_functions/.dockerignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_functions/.dockerignore:3]: Oh! My bad! I did not read it carefully.

File: oak_functions/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_functions/Dockerfile:17]: Of course :D

File: scripts/deploy_oak_functions_loader (hunk 1 above)
Comment (rbehjati) [reply scripts/deploy_oak_functions_loader:24]: I did not realize that `gcp_common` is checking this.

**Review by rbehjati — APPROVED** [review]
Thanks :)

**Pull request merged**